File: SchoolWebApp-API/Project.Core/Entities/CBE/Cocurriculum/StudentCoCurriculumActivity.cs
using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Students;

namespace SchoolWebApp.Core.Entities.CBE.Cocurriculum
{
    public class StudentCoCurriculumActivity: Base
    {
        public int StudentId { get; set; }
        public Student? Student { get; set; } = null;
        public int CoCurriculumActivityId { get; set; }
        public CoCurriculumActivity? CoCurriculumActivity { get; set; } = null;
        public string? Description { get; set; } = null;
        public List<StudentCoCurriculumScore> StudentCoCurriculumScores { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/cbe/cocurriculum/models/student-co-curriculum-activity.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class StudentCoCurriculumActivity extends ResourceModel<StudentCoCurriculumActivity> {
    public studentId?: number;
    public student?: any;
    public coCurriculumActivityId?: number;
    public coCurriculumActivity?: any;
    public description?: string;

    constructor(model?: Partial<StudentCoCurriculumActivity>) {
        super(model);
    }
}
